testing/describes.js

```diff
@@ -304,53 +306,41 @@ function describeEnv(factory) {
    * @param {function(string, function())} describeFunc
    */
   const templateFunc = function (name, spec, fn, describeFunc) {
-    const fixtures = [new SandboxFixture(spec)];
-    factory(spec).forEach((fixture) => {
-      if (fixture && fixture.isOn()) {
-        fixtures.push(fixture);
-      }
-    });
-    return describeFunc(name, function () {
+    const fixtures = [new SandboxFixture(spec)].concat(
+      factory(spec)
+        .filter(Boolean)
+        .filter((fixture) => fixture.isOn())
+    );
+
+    return describeFunc(name, () => {
       const env = Object.create(null);
 
-      beforeEach(() => {
-        let totalPromise = undefined;
-        // Set up all fixtures.
-        fixtures.forEach((fixture, unusedIndex) => {
-          if (totalPromise) {
-            totalPromise = totalPromise.then(() => fixture.setup(env));
-          } else {
-            const res = fixture.setup(env);
-            if (res && typeof res.then == 'function') {
-              totalPromise = res;
-            }
-          }
-        });
-        return totalPromise;
+      beforeEach(async () => {
+        for (let i = 0; i < fixtures.length; ++i) {
+          await fixtures[i].setup(env);
+        }
```

Comment:
Nice clean up! The old code was added 4 years ago in https://github.com/ampproject/amphtml/pull/5569/files#diff-8f8103e0f4e0052eab727bfd5c21e090R138, back when we were using node v4.